datastream-utils/src/main/java/com/linkedin/datastream/common/FutureUtils.java

```diff
@@ -0,0 +1,43 @@
+/**
+ *  Copyright 2019 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.common;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+
+public class FutureUtils {
+
+  private static final ScheduledExecutorService SCHEDULED_EXECUTOR_SERVICE = Executors.newScheduledThreadPool(100);
+
+  /**
+   * This function waits till timeout for Future to complete, otherwise, it cancels it.
+   * @param futureToAwait Future to complete/stop
+   * @param timeout time to wait before aborting future
+   * @return CompletableFuture for this operation
```

Comment:
The javadocs can be improved for greater accuracy so users don't have to grok the code themselves.

- The description can be something like: 
```java
/**
 * Check if {@code future} is done after {@code timeout} ... (and the rest of the story)
```
- `Future to complete/stop` - check/cancel?
- `before aborting future` - before checking actually
- We'd need to document when and how the returned future is completed successfully, completed w/ an exception, or cancelled.